pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -185,6 +224,61 @@ func newLibvirtDomainManager(connection cli.Connection, virtShareDir string, not
 	return &manager, nil
 }
 
+func detectEFIROMS(arch, ovmfPath string) *efiConfiguration {
+	if arch == "arm64" {
+		var codeArm64, varsArm64 string
+
+		_, err := os.Stat(filepath.Join(ovmfPath, EFICodeAARCH64))
+		if err == nil {
+			codeArm64 = EFICodeAARCH64
+		}
+		_, err = os.Stat(filepath.Join(ovmfPath, EFIVarsAARCH64))
+		if err == nil {
+			varsArm64 = EFIVarsAARCH64
+		}
+
+		return &efiConfiguration{
+			ovmfPath: ovmfPath,
+			code:     codeArm64,
+			vars:     varsArm64,
+		}
+	}
+
+	// detect EFI with SecureBoot
+	var codeWithSB, varsWithSB string
+	_, err := os.Stat(filepath.Join(ovmfPath, EFICodeSecureBoot))
+	if err == nil {
+		codeWithSB = EFICodeSecureBoot
+	}
+	_, err = os.Stat(filepath.Join(ovmfPath, EFIVarsSecureBoot))
+	if err == nil {
+		varsWithSB = EFIVarsSecureBoot
+	}
+
+	// detect EFI without SecureBoot
+	var code, vars string
+	_, err = os.Stat(filepath.Join(ovmfPath, EFICode))
+	if err == nil {
+		code = EFICode
+	} else {
+		// The combination (EFICodeSecureBoot + EFIVars) is valid
+		// for booting in EFI mode with SecureBoot disabled
+		code = codeWithSB
+	}
+	_, err = os.Stat(filepath.Join(ovmfPath, EFIVars))
+	if err == nil {
+		vars = EFIVars
+	}
+
+	return &efiConfiguration{
+		ovmfPath:       ovmfPath,
```

Comment:
Consider elminating the `ovmfPath` member completely and just directly saving the discovered path.